### ISSUE TITLE
fish: Could not locate end of block. The “end” command is missing, misspelled or a “;” is missing.

### ISSUE THREAD

**Issue by jpjp** [fish-shell/fish-shell#2075]
```
$ fish
$ mkdir My\ Files
$ cd "My<tab>
~> cd "Myfish: Could not locate end of block. The “end” command is missing, misspelled or a “;” is missing.
/usr/share/fish/functions/__fish_complete_cd.fish (line 2): begin; printf "%s\n" "My*/ 
                                                            ^
in . (source) call of file “-”,
	called on line 35 of file “/usr/share/fish/functions/__fish_complete_cd.fish”,

in function “__fish_complete_cd”,
	called on standard input,

in command substitution
	called on standard input,

source: Error while reading file “-”
 Files/fish: Could not locate end of block. The “end” command is missing, misspelled or a “;” is missing.
/usr/share/fish/functions/__fish_complete_cd.fish (line 2): begin; printf "%s\n" "My Files/*/ 
                                                            ^
in . (source) call of file “-”,
	called on line 35 of file “/usr/share/fish/functions/__fish_complete_cd.fish”,

in function “__fish_complete_cd”,
	called on standard input,

in command substitution
	called on standard input,

source: Error while reading file “-”
```

**Comment by jpjp** [comment]
fish-2.1.1-3.fc22.x86_64

**Comment by ridiculousfish** [comment]
Same sort of issue as #562, but it's surprising we haven't noticed this variant (with quotes) until now. I wonder if this has gotten worse. Let me regress.

**Comment by ridiculousfish** [comment]
It looks like this was a regression in fish 2.0.

**Comment by ridiculousfish** [comment]
It looks like maybe the issue existed in fish 1.x but the error message was suppressed.

**Comment by jpjp** [comment]
Here is the downstream issue: https://bugzilla.redhat.com/show_bug.cgi?id=1222772
Maybe they can push an update once it's working again.

**Comment by ridiculousfish** [comment]
Bisects to 07f5319472ed7c052b25aec7d29e92126b9bca48

**Comment by ridiculousfish** [comment]
I think this was introduced aeons ago in 376e199e, when I accidentally replaced an unescape with an escape. Fixed as 2008415106144a3809b3b84a40f551d2ffe39472 . Thanks for reporting this!!

**Comment by jpjp** [comment]
No problem! You are very responsive to bug reports, will definitely do this again :)
